### PR TITLE
Menu: fix router NavigationDuplicated error when using vue-router@^3.1.0

### LINKED ISSUE #17044: [Bug Report] 升级vue-router至3.1以后版本，导航组件重复点击报错 NavigationDuplicated
### Element UI version
2.11.1

### OS/Browsers version
Win/chrome

### Vue version
2.6.10

### Reproduction Link
https://github.com/vuejs/vue-router/issues/2881

### Steps to reproduce
升级vue-router至新版本后，使用 el-menu 来导航页面，重复点击某一个导航项就会在控制台报错，vue-router官方给出了一些修改方法，element这边是不是要同步修改

### What is Expected?
不报错 NavigationDuplicated

### What is actually happening?
控制台 NavigationDuplicated

### PR DESCRIPTION
**Pull request by iamkun**
Please make sure these boxes are checked before submitting your PR, thank you!
close #17044 

relevant #17145

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: packages/menu/src/menu.vue [hunk 1]
```diff
@@ -294,7 +294,10 @@
       routeToItem(item, onError) {
         let route = item.route || item.index;
         try {
-          this.$router.push(route, () => {}, onError);
+          const result = this.$router.push(route, () => {}, onError)
+          // vue-router 3.1.0+ push/replace cause NavigationDuplicated error (In a Promise)
+          // https://github.com/ElemeFE/element/issues/17044
+          result.catch && result.catch(() => {});
```
Comment (a631807682) [inline packages/menu/src/menu.vue]: `push`有第二第三个参数的时候不会返回`Promise`，之前问题是`onError`会直接打印`error`。

Comment (iamkun) [reply packages/menu/src/menu.vue]: 谢谢 看了下代码 是这样的

又调整了一下 fix 现在这样应该是影响最小的了

Comment (a631807682) [reply packages/menu/src/menu.vue]: 应该没问题

**Comment by Rhyheart** [comment]
为什么我升级完还是一样的问题
image

**Comment by gdmec07150725** [comment]
I had upgrade element to 2.13.0 but the issue still exist! The vue-router version in my project is 3.1.5

**Comment by yanzhen0610** [comment]
https://github.com/vuejs/vue-router/blob/dfc289202703319cf7beb38d03c9258c806c4d62/src/util/errors.js#L23

The error in vue router is updated, so it cannot be caught in this way in current version(3.3.4).

**Pull request merged**